File: covid_vac.py
# This documentation pulls out the necessary columns from the CDC COVID-19 Vaccinations CSV File.

# In order to install the necessary libraries, run the following commands in your computer terminal.
# These commands were run using a MacOS computer terminal and Python 3 and may need to be adjusted based on your interface. 
  # First, install pip
    #curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
    #python3 get-pip.py
  # Then, install pandas
    #pip3 install pandas

# Import the necessary libraries
import csv
import pandas as pd

# Read the COVID-19 Vaccination CSV directly into data frame using pandas
covid_vac = pd.read_csv('COVID-19_Vaccinations_in_the_United_States_Jurisdiction.csv')

# Print column name and index
for idx,column in enumerate(covid_vac.columns):
  print(idx,column)
#important columns
  #[0] - Date
  #[2] - Location: gives state
  #Vaccine Distributions
    #[3] - General
    #[4] - Janssen
    #[5] - Moderna
    #[6] - Pfizer
  #Completed Series by Drug
    #[33] - General
    #[41] - Janssen
    #[42] - Moderna
    #[43] - Pfizer
  #Additional Doses
    #[57] - General?
    #[69] - Janssen
    #[67] - Moderna
    #[68] - Pfizer
  #Do we want to look by age?

data = covid_vac.iloc[:,[0, 2, 3, 4, 5, 6, 33, 41, 42, 43, 57, 67, 68, 69]]
csv_data = data.to_csv("/Users/rhea/Desktop/STAT_Project/STAT_Project/covid_vac.csv", index = False)

File: covid_vac.R
#I got confused downloading Pandas so I pulled out the columns using dplyr

#subset covid vaccine dataset using dplyr (not as familiar with Python's Pandas)

#read in dplyr package
library(dplyr)

#read in COVID-19 Vaccination data
cov_vac <- read.csv("covid_vac.csv")

#subset to selected rows (same as is the python script but by column name)
cov_vac <- cov_vac %>% select(Date, Location, Distributed, Distributed_Janssen, Distributed_Moderna, Distributed_Pfizer, Series_Complete_Yes, 
         Series_Complete_Janssen, Series_Complete_Moderna, Series_Complete_Pfizer, Additional_Doses, Additional_Doses_Janssen,
         Additional_Doses_Moderna, Additional_Doses_Pfizer)

View(cov_vac)

#done!
